### PR TITLE
listener: deprecate envoy_reloadable_features_enable_update_listener_socket_options

### LINKED ISSUE #29093: envoy_reloadable_features_enable_update_listener_socket_options deprecation
Your change #23189 (listener: support update socket_options) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by soulxu**
Commit Message: listener: deprecate envoy_reloadable_features_enable_update_listener_socket_options
Additional Description:
Risk Level: low
Testing: unitetest
Docs Changes: n/a
Release Notes: remove runtime flag
Platform Specific Features: n/a
Fixes #29093

### CODE REVIEW

File: changelogs/current.yaml
Comment (adisuissa) [inline changelogs/current.yaml:146]: nit: 
```suggestion
   Removed ``envoy.reloadable_features.enable_update_listener_socket_options`` runtime flag and legacy code paths.
```

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!
Left a small nit regarding the release note.

/retest

File: changelogs/current.yaml
Comment (soulxu) [reply changelogs/current.yaml:146]: got it, thanks!

File: changelogs/current.yaml
Comment (adisuissa) [inline changelogs/current.yaml:146]: nit:
```suggestion
   Removed ``envoy.reloadable_features.enable_update_listener_socket_options`` runtime flag and legacy code paths.
```

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

Assigning Alyssa for senior-maintainer signoff.
/assign @alyssawilk

**Review by alyssawilk — DISMISSED** [review]
LGTM modulo Adi's request.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**
